frontends/p4/uniqueNames.h

```diff
@@ -80,8 +80,14 @@ class FindSymbols : public Inspector {
     }
     void postorder(const IR::Declaration_Variable* decl) override
     { doDecl(decl); }
-    void postorder(const IR::Declaration_Constant* decl) override
-    { doDecl(decl); }
+    void postorder(const IR::Declaration_Constant* decl) override {
+        // Skip toplevel constants with names like __
+        // We assume that these do not clash and no new symbols with
+        // these names will be added.
+        if (decl->getName().name.startsWith("__") && getParent<IR::P4Program>())
```

Comment:
Could this cause an issue with inlining and constants that have been declared with __?

Reply:
I don't think we ever move objects into the global namespace currently, so I don't think this is an issue with the current implementation. You are right that in the future this may cause problems. The right solution is to reserve some names in the spec for internal use of the compiler, and then we can guarantee that this is safe. I could make this safer by checking explicitly just for __v1model_version, although that won't work for other architectures. Maybe we should discuss this in the LDWG.